File: chor-extraction/src/main/java/ast/sp/nodes/ProcedureDefinitionSP.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;

public class ProcedureDefinitionSP implements Behaviour {
    private final String procedure;
    private final Behaviour behaviour;

    public ProcedureDefinitionSP(String procedure, Behaviour behaviour) {
        this.procedure = procedure;
        this.behaviour = behaviour;
    }

    public String getProcedure() {
        return procedure;
    }

    public Behaviour getBehaviour() {
        return behaviour;
    }

    public String toString() {
        return "def " + procedure + " =" + behaviour.toString();
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/ProcedureDefinitionSP.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour

data class ProcedureDefinitionSP(val behaviour: Behaviour) : Behaviour {

    var visited: Boolean
    init {
        visited = false
    }

    override fun toString(): String {
        return behaviour.toString()
    }

    override fun findRecProcCall(procname: String): Boolean {
        return behaviour.findRecProcCall(procname)
    }

    override fun copy(): ProcedureDefinitionSP {
        return ProcedureDefinitionSP(behaviour.copy())
    }
}
